### PR TITLE
add inverse_block_masking

### PR DESCRIPTION
**Pull request by Manishram-ai**
Added inverse_block_masking to masking.py

I tried to keep it like pytorch code using nn.Module. This code now looks similar to other masking implementation. 

Thanks!

Edit:

- Improved inverse masking that  handles wrap-around, different aspect ratios, and makes sure there are enough visible patches in the block mask
- Also added import statement for math, random and einops
- Class InverseBlockMasking nn.Module is a wrapper to match other masking modules

### CODE REVIEW

**Comment by clane9** [comment]
closing, stale
